dessia_common/workflow/core.py

```diff
@@ -1941,6 +1943,119 @@ def _method_jsonschemas(self):
         warnings.warn("method_jsonschema method is deprecated. Use method_schema instead", DeprecationWarning)
         return self.method_schemas
 
+    @staticmethod
+    def generate_object_input_string(objects):
+        """ Generate input string for object(s). """
+        signature = "('Set your arguments here')"
+        if len(objects) > 1:
+            instances = [f"\n\t{o.__class__.__name__}{signature}" for o in objects]
+            return f"[{','.join(instances)}\n]"
+        return f"{objects[0].__class__.__name__}{signature}"
+
+    @staticmethod
+    def generate_file_input_string(type_files):
+        """ Generate input string for file(s). """
+        signature = ".from_file('Set your filepath here')"
+        if len(type_files) > 1:
+            instances = [f"\n\t{o.__class__.__name__}{signature}" for o in type_files]
+            return f"[{','.join(instances)}\n]"
+        return f"{type_files[0].__class__.__name__}{signature}"
+
+    @staticmethod
+    def is_object_sequence(sequence):
+        """ Checks if a sequence contains serializable objects. """
+        return any(isinstance(element, SerializableObject) for element in sequence)
+
+    def process_sequence(self, sequence):
+        """ Processes a sequence based on its content. """
+        if self.is_object_sequence(sequence):
+            return self.generate_object_input_string(sequence)
+        if is_file_or_file_sequence(sequence):
+            return self.generate_file_input_string(sequence)
+        return repr(sequence)
+
+    def _split_input_values(self):
+        """ Split the input values for blocks and non-block variables. """
+        nbv_values = [self.input_values[i] for i, v in enumerate(self.workflow.inputs) if
+                      v in self.workflow.nonblock_variables]
+        block_values = [self.input_values[i] for i, v in enumerate(self.workflow.inputs) if
+                        v not in self.workflow.nonblock_variables]
+        return nbv_values, block_values
+
+    @staticmethod
+    def _update_imports(input_type, input_name):
+        """ Update imports based on the schema of the input. """
+        schema = get_schema(annotation=input_type, attribute=SchemaAttribute(input_name))
+        imports = schema.get_import_names(import_names=[])
+        return imports
```

Comment:
You could directly `return schema.get_import_names(import_names=[])`

Reply:
done !